### PR TITLE
Waiting participants hook

### PR DESCRIPTION
**Pull request by harshithpabbati**
This PR uses `useWaitingParticipants` hooks.

### CODE REVIEW

**Review by nienkedekker — APPROVED** [review]

File: custom/shared/contexts/WaitingRoomProvider.js [hunk 1]
```diff
@@ -1,99 +1,31 @@
...
-    }
-    updateWaitingParticipant(id, true);
-  };
-  const denyAccess = (id = 'all') => {
-    if (id === 'all') {
-      updateAllWaitingParticipants(false);
-      return;
-    }
-    updateWaitingParticipant(id, false);
-  };
```
Comment (jessmitch42) [inline custom/shared/contexts/WaitingRoomProvider.js:95]: so much shorter 😍

Comment (jessmitch42) [inline custom/shared/contexts/WaitingRoomProvider.js:95]: so much shorter 😍

**Review by jessmitch42 — CHANGES_REQUESTED** [review]
Looking good! Loving these changes split into smaller PRs. Thanks for doing that!

A couple bugs I noticed while testing: 

- clicking "Go back" doesn't seem to do anything if the room isn't valid
CleanShot 2022-04-07 at 15 29 05@2x

- I get the participant waiting to join notification as an owner but when I let them join I get this error on the owner and participant's end
CleanShot 2022-04-07 at 15 30 59@2x

File: custom/basic-call/pages/not-found.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 import MessageCard from '@custom/shared/components/MessageCard';
+import { useRouter } from 'next/router';
 
 export default function RoomNotFound() {
+  const router = useRouter();
+
   return (
     <div className="not-found">
-      <MessageCard error header="Room not found">
+      <MessageCard error header="Room not found" onBack={() => router.push('/')}>
```
Comment (jessmitch42) [inline custom/basic-call/pages/not-found.js:10]: this anonymous`onBack` function can be a named function (I've given this feedback on other PRs, please use it as a general request for React demo code.)

`const returnToHomePage = () => router.push('/')`
`onBack={returnToHomePage}`

**Comment by jessmitch42** [comment]
I noticed you opened a new PR for a piece of feedback in this PR. Small request: please either link to related PRs in the PR comment where the feedback was added so I know where to look or just add the changes here. 🙏  I don't spend enough time in this repo to keep track from memory. (Your fix solved the bug, though 👍 )

**Review by jessmitch42 — APPROVED** [review]
Looks great!

**Pull request merged**